code/extensions/Lumberjack.php

```diff
@@ -35,30 +35,48 @@ public function getExcludedSiteTreeClassNames() {
 	 * @param FieldList $fields
 	 */
 	public function updateCMSFields(FieldList $fields) {
-		$excluded = $this->owner->getExcludedSiteTreeClassNames();
-		if(!empty($excluded)) {
-			$pages = SiteTree::get()->filter(array(
-				'ParentID' => $this->owner->ID,
-				'ClassName' => $excluded
-			));
+		$childPages = $this->getLumberjackChildPages();
+
+		if($childPages !== null) {
 			$gridField = new GridField(
-				"ChildPages",
+				'ChildPages',
 				$this->getLumberjackTitle(),
-				$pages,
+				$childPages,
 				$this->getLumberjackGridFieldConfig()
 			);
 
 			$tab = new Tab('ChildPages', $this->getLumberjackTitle(), $gridField);
+
 			$fields->insertAfter($tab, 'Main');
 		}
 	}
 
+	/**
+	 * @return null|DataList
+	 */
+	protected function getLumberjackChildPages() {
+		if(method_exists($this->owner, 'getLumberjackChildPages')) {
+			return $this->owner->getLumberjackChildPages();
+		}
+
+		$excluded = $this->owner->getExcludedSiteTreeClassNames();
+
+		if(!empty($excluded)) {
+			return SiteTree::get()->filter(array(
+				'ParentID' => $this->owner->ID,
+				'ClassName' => $excluded
+			));
+		}
+
+		return null;

```

Comment:
Can we change this to ArrayList instead and change the return type to SS_List (interface)?